File: backend/repositories/health_repository.py
from .base_repository import BaseRepository


class HealthRepository(BaseRepository):
    """Repository for health check operations (DB connectivity tests)"""

    def check_database_connection(self) -> dict:
        """
        Test connection to SQL Server by executing a simple query
        Returns: Dict with connection status and server info
        """
        try:
            query = "SELECT @@VERSION as version, GETDATE() as server_time"
            result = self.execute_query(query)

            if result:
                row = result[0]
                return {
                    "status": "connected",
                    "server_version": row[0],
                    "server_time": str(row[1])
                }
            else:
                return {"status": "error", "message": "Query executed but no results"}

        except Exception as e:
            return {
                "status": "error",
                "message": str(e)
            }

File: backend-go/repositories/health_repository.go
package repositories

import (
	"database/sql"
	"fmt"
)

// HealthRow holds the SQL Server diagnostics returned by a connectivity check.
type HealthRow struct {
	Version    string
	ServerTime string
}

// HealthRepository provides health-check operations (DB connectivity tests).
type HealthRepository struct {
	DB *sql.DB
}

// NewHealthRepository wires the shared connection pool into the repository.
func NewHealthRepository(db *sql.DB) *HealthRepository {
	return &HealthRepository{DB: db}
}

// CheckConnection verifies SQL Server connectivity and returns its version and
// current time. The timestamp is converted to a string directly in SQL so it
// can be scanned into a plain string.
func (r *HealthRepository) CheckConnection() (HealthRow, error) {
	const query = "SELECT @@VERSION, CONVERT(varchar, GETDATE(), 120)"

	row := r.DB.QueryRow(query)

	var health HealthRow
	if err := row.Scan(&health.Version, &health.ServerTime); err != nil {
		return HealthRow{}, fmt.Errorf("database health check failed: %w", err)
	}

	return health, nil
}
